x/ibc/light-clients/07-tendermint/types/store.go

```diff
@@ -35,3 +40,50 @@ func GetConsensusState(store sdk.KVStore, cdc codec.BinaryMarshaler, height expo
 
 	return consensusState, nil
 }
+
+// IterateProcessedTime iterates through the prefix store and applies the callback.
+// If the cb returns true, then iterator will close and stop.
+func IterateProcessedTime(store sdk.KVStore, cb func(key, val []byte) bool) {
+	iterator := sdk.KVStorePrefixIterator(store, []byte(host.KeyConsensusStatePrefix))
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		keySplit := strings.Split(string(iterator.Key()), "/")
+		// processed time key in prefix store has format: "consensusState/<height>/processedTime
+		if len(keySplit) != 3 {
```

Comment:
are there other keys with 3 elements? should we double check that the last element is `"processedTime"`

Reply:
No, because we prefixed with consensus state key first. So nothing else should be in here but consensus states and processed times. But I can add that double check just for sanity reasons

Reply:
+1 for the sanity checks. If anyone copies this code and adds another consensus state key for metadata then this could cause an issue